kpt-backstage-plugins/plugins/cad/src/components/ResourceEditorDialog/components/FirstClassEditors/ApplyReplacementsEditor/ApplyReplacementsEditor.tsx

```diff
@@ -0,0 +1,471 @@
+import { SelectItem } from '@backstage/core-components';
+import { TextField } from '@material-ui/core';
+import { dump, load } from 'js-yaml';
+import React, { ChangeEvent, Fragment, useEffect, useState } from 'react';
+import {
+  ApplyReplacement,
+  ReplacementOptions,
+  ReplacementResourceSelector,
+  ReplacementSource,
+  ReplacementTarget,
+} from '../../../../../types/ApplyReplacement';
+import {
+  KubernetesKeyValueObject,
+  KubernetesResource,
+} from '../../../../../types/KubernetesResource';
+import { PackageResource } from '../../../../../utils/packageRevisionResources';
+import { sortByLabel } from '../../../../../utils/selectItem';
+import { Select } from '../../../../Controls/Select';
+import { EditorAccordion } from '../Controls/EditorAccordion';
+import { KeyValueEditorAccordion } from '../Controls/KeyValueEditorAccordion';
+import { SingleTextFieldAccordion } from '../Controls/SingleTextFieldAccordion';
+import { useEditorStyles } from '../styles';
+
+type OnUpdatedYamlFn = (yaml: string) => void;
+
+type ApplyReplacementsEditorProps = {
```

Comment:
@natasha41575 can you please look at the editor logic for ApplyReplacements here?

Reply:
Looking

Reply:
Overall it LGTM